src/main/java/net/mcreator/ui/minecraft/states/block/JBlockPropertiesListEntry.java

```diff
@@ -0,0 +1,166 @@
+			List<JBlockPropertiesListEntry> propertiesList) {
+		super(new BorderLayout(20, 5));
+		this.mcreator = listPanel.getMCreator();
+
+		setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+
+		nameLabel.setToolTipText(nameLabel.getText());
+		nameLabel.setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+		nameLabel.setPreferredSize(new Dimension(0, 28));
+		ComponentUtils.deriveFont(nameLabel, 16);
+
+		JPanel namePane = new JPanel(new BorderLayout());
+		namePane.setOpaque(false);
+		namePane.add("North", HelpUtils.wrapWithHelpButton(gui.withEntry("block/custom_property_name"),
+				L10N.label("elementgui.block.custom_property.name")));
+		namePane.add("Center", nameLabel);
+		namePane.setPreferredSize(new Dimension(240, 0));
+
+		typeLabel.setBorder(BorderFactory.createEmptyBorder(0, 5, 0, 5));
+		typeLabel.setPreferredSize(new Dimension(0, 28));
+		ComponentUtils.deriveFont(typeLabel, 16);
+
+		JPanel typePane = new JPanel(new BorderLayout());
+		typePane.setOpaque(false);
+		typePane.add("North", HelpUtils.wrapWithHelpButton(gui.withEntry("block/custom_property_type"),
+				L10N.label("elementgui.block.custom_property.type")));
+		typePane.add("Center", typeLabel);
+		typePane.setPreferredSize(new Dimension(160, 0));
+
+		boundsPane = new JPanel(new BorderLayout());
+		boundsPane.setOpaque(false);
+		boundsPane.setPreferredSize(new Dimension(350, 0));
+		boundsPane.add("North", HelpUtils.wrapWithHelpButton(gui.withEntry("block/custom_property_values"),
+				L10N.label("elementgui.block.custom_property.values")));
+
+		defValuePane.setOpaque(false);
+		defValuePane.add("North", HelpUtils.wrapWithHelpButton(gui.withEntry("block/custom_property_default_value"),
+				L10N.label("elementgui.block.custom_property.default")));
+
+		JPanel line = new JPanel(new BorderLayout());
+		line.setLayout(new BoxLayout(line, BoxLayout.X_AXIS));
+		line.add(namePane);
+		line.add(typePane);
+		line.add(boundsPane);
+		line.add(defValuePane);
+
+		remove.setText(L10N.t("elementgui.block.custom_property.remove"));
+		remove.addActionListener(e -> listPanel.removeProperty(this));
+		add("West", line);
+		add("East", PanelUtils.pullElementUp(remove));
+
+		propertiesList.add(this);
+		propertyEntries.add(this);
+		propertyEntries.revalidate();
+		propertyEntries.repaint();
+	}
+
+	@Override public void setEnabled(boolean enabled) {
+		super.setEnabled(enabled);
+
+		remove.setEnabled(enabled);
+	}
+
+	public PropertyData<?> getPropertyData() {
+		return data;
+	}
+
+	@SuppressWarnings("unchecked") public Block.PropertyEntry getEntry() {
+		Block.PropertyEntry entry = new Block.PropertyEntry();
+		entry.data = new PropertyDataWithValue<>((PropertyData<Object>) data, data.getValue(defaultValue));
+		return entry;
+	}
+
+	public void setEntry(Block.PropertyEntry entry) {
+		setProperty(entry.data);
+	}
+
+	public void setProperty(PropertyDataWithValue<?> prop) {

```

Comment:
Does this way to pass entries look OK-ish?

Reply:
Closing as this will be soon outdated